.github/workflows/check-release.yml

```diff
@@ -0,0 +1,117 @@
+          done
+
+      - name: Set outputs
+        id: version_check
+        run: |
+          echo "new_release=${{ env.NEW_RELEASE }}" >> $GITHUB_OUTPUT
+          echo "package_name=${{ env.PACKAGE_NAME }}" >> $GITHUB_OUTPUT
+          echo "package_version=${{ env.PACKAGE_VERSION }}" >> $GITHUB_OUTPUT
+    outputs:
+      new_release: ${{ steps.version_check.outputs.new_release }}
+      package_name: ${{ steps.version_check.outputs.package_name }}
+      package_version: ${{ steps.version_check.outputs.package_version }}
+
+  build-publish:
+    needs: [check-version]
+    if: |
+      needs.check-version.outputs.new_release == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install hatch
+      - name: Extract release notes from ${{needs.check-version.outputs.package_name}}/CHANGELOG.md
+        id: extract
+        run: |
+          formatted_date=$(date +"%Y-%m-%d")
+          python tools/extract_release_notes.py \
+          "${{needs.check-version.outputs.package_name}}/CHANGELOG.md" \
+          "${{needs.check-version.outputs.package_version}} — $formatted_date"
+      - name: Tag main and create release on github
+        run: |
+          ./tools/release.sh mckinsey vizro ${{ secrets.GITHUB_TOKEN }} \
+          ${{needs.check-version.outputs.package_name}} \
+          ${{needs.check-version.outputs.package_version}} release_body.txt
+      - name: Build package
+        run: |
+          cd "${{ needs.check-version.outputs.package_name }}"
+          hatch build
+      - name: Set PyPI token
+        run: |
+          if [ "${{ needs.check-version.outputs.PACKAGE_NAME }}" == "vizro-core" ]; then
+              echo 'PYPI_TOKEN=${{ secrets.VIZRO_PYPI_TOKEN }}' >> $GITHUB_ENV
+          fi
+      - name: Publish distribution 📦 to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          packages-dir: ${{ needs.check-version.outputs.package_name }}/dist
+          password: ${{ env.PYPI_TOKEN }}
+
+  version-bump:
+    needs: [check-version]
+    if: |
+      needs.check-version.outputs.new_release == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install hatch
+      - name: Bump version to next patch
+        run: |
+          formatted_date=$(date +"%Y-%m-%d/%H-%M-%S")
+          cd "${{ needs.check-version.outputs.package_name }}"
+          git checkout -b "release/version_bump_next_minor/${formatted_date}"
+          hatch version patch,dev
```

Comment:
And what if we want to release a `minor` version rather than a `patch` version?

Reply:
Then we have to change the version manually, but this is likely happening only when a breaking PR hits main, so it should be done with that PR instead of right after release of the previous release